lib/improver/blending/weighted_blend.py

```diff
@@ -454,165 +454,216 @@ def process(self, cube, weights=None):
-                    orig_cell_methods = cube_thres.cell_methods
-                    # Calculate the weighted average.
-                    cube_new = cube_thres.collapsed(self.coord,
-                                                    iris.analysis.MEAN,
-                                                    weights=weights_array)
-                    # Update the name of the cell_method created by Iris to
-                    # 'weighted_mean' to be consistent.
-                    new_cell_methods = cube_new.cell_methods
-                    extra_cm = (set(new_cell_methods) -
-                                set(orig_cell_methods)).pop()
-                    add_renamed_cell_method(cube_new,
-                                            extra_cm,
-                                            'mean')
-
-                # Else use the maximum probability aggregator.
-                elif self.mode == "weighted_maximum":
-                    # Set equal weights if none are provided
-                    if weights is None:
-                        num = len(cube_thres.coord(self.coord).points)
-                        weights = np.ones(num) / float(num)
-                    # Set up aggregator
-                    MAX_PROBABILITY = (Aggregator(
-                        'maximum',  # Use CF-compliant cell method.
-                        MaxProbabilityAggregator.aggregate))
-
-                    cube_new = cube_thres.collapsed(self.coord,
-                                                    MAX_PROBABILITY,
-                                                    arr_weights=weights)
-                cube_new = conform_metadata(
-                    cube_new, cube_thres, coord=self.coord,
-                    cycletime=self.cycletime,
-                    coords_for_bounds_removal=self.coords_for_bounds_removal)
-                cubelist.append(cube_new)
-            result = cubelist.merge_cube()
-            if isinstance(cubelist[0].data, np.ma.core.MaskedArray):
-                result.data = np.ma.array(result.data)
-        # If set adjust values of collapsed coordinates.
-        if self.coord_adjust is not None:
-            for crd in result.coords():
-                if cube.coord_dims(crd.name()) == coord_dim:
-                    pnts = cube.coord(crd.name()).points
-                    crd.points = np.array(self.coord_adjust(pnts),
-                                          dtype=crd.points.dtype)
-
-        return result
+                slices_over_threshold = [cube]
+        return slices_over_threshold
+
+    def _equal_weights(self, cube):
+        num = len(cube.coord(self.coord).points)
+        weights = np.ones(num) / float(num)
+        return weights
+
+    def _blend_weighted_maximum(self, cube_thres, weights):
+        # Set equal weights if none are provided
+        if weights is None:
+            weights = self._equal_weights(cube_thres)
+        # Set up aggregator
+        MAX_PROBABILITY = (Aggregator(
```

Comment:
Aggregator set up too many times?

Reply:
I agree. It's better to inject the dependency in my opinion, but I didn't want to do that in this PR.